Sources/Common/Communication/EventBusHandler.swift

```diff
@@ -1,66 +1,87 @@
 import Combine
 import Foundation
 
+/// Manages the handling of events through an event bus and coordinates with event storage.
+/// It handles both in-memory and file-based storage of events.
+// sourcery: InjectRegisterShared = "EventBusHandler"
+// sourcery: InjectSingleton
 public class EventBusHandler {
     private let eventBus: EventBus
     private let eventStorage: EventStorage
+    private var memoryStorage: [String: [any EventRepresentable]] = [:]
 
+    /// Initializes the EventBusHandler with an EventBus and EventStorage.
+    /// Automatically loads events from file-based storage into in-memory storage upon initialization.
     public init(eventBus: EventBus, eventStorage: EventStorage) {
         self.eventBus = eventBus
         self.eventStorage = eventStorage
+        loadEventsFromStorage()
     }
 
-    public func send<E>(_ event: E) where E: EventRepresentable {
-        if eventBus.send(event) == false {
-            let key = event.key
+    /// Loads events from file-based persistent storage into in-memory storage for each registered event type.
+    /// This facilitates quick access and ensures events can be replayed to late subscribers.
+    private func loadEventsFromStorage() {
+        EventTypesRegistry.allEventTypes().forEach { eventType in
             do {
-                try eventStorage.store(event: event, forKey: key)
+                let key = eventType.key
+                let events: [any EventRepresentable] = try eventStorage.loadEvents(ofType: eventType)
+                memoryStorage[key, default: []].append(contentsOf: events)
             } catch {
-                print("Error storing event: \(error)")
+                // Handle the error, for example, log it or take some recovery actions
+                print("Error loading events for \(eventType): \(error)")
             }
```

Comment:
I agree, logging errors when a store encounters an error is a great idea. 

I do have 2 suggestions about this. 
1. Instead of `print`, could we use `logger.error` so that customers can more easily see these errors and report them to us? 
2. Could we modify `eventStorage.loadEvents(ofType: eventType)` to not `throw`? I prefer to [encapsulate all error catching in 1 place such as we do here in the SDK](https://github.com/customerio/customerio-ios/blob/94cbf686c3c2a405534cfe908f7166558a3e0b5d/Sources/Common/Util/FileStorage.swift#L93-L102). I like this approach because it requires less boilerplate code (having to use try/catch/log each time we call `eventStorage.loadEvents`.